src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -585,6 +610,65 @@ def alembic(self) -> Alembic:
             raise ValueError("Store not initialized")
         return self._alembic
 
+    @classmethod
+    def filter_and_paginate(
+        cls,
+        session: Session,
+        query: Union[Select[AnySchema], SelectOfScalar[AnySchema]],
+        table: Type[AnySchema],
+        filter_model: FilterBaseModel,
+    ) -> Page[B]:
+        """Given a query, return aPage instance with a list of filtered Models.
+
+        Args:
+            session: The SQLModel Session
+            query: The query to execute
+            table: The table to select from
+            filter_model: The filter to use, including pagination and sorting
+
+        Returns:
+            The Domain Model representation of the DB resource
+        """
+
+        # Filtering
+        filters = filter_model.generate_filter(table=table)
+
+        if filters is not None:
+            query = query.where(filters)
+
+        # Sorting
+        query = query.order_by(getattr(table, filter_model.sort_by))
```

Comment:
if you move this down you don'y need to set `query.order_by(None)` below I think